File: libs/service_layer/search/SearchEngine/src/main/kotlin/net/dankito/service/search/specific/FilesSearch.java
package net.dankito.service.search.specific;


import net.dankito.deepthought.model.FileLink;
import net.dankito.service.search.Search;
import net.dankito.service.search.SearchCompletedListener;

import java.util.Collection;


public class FilesSearch extends Search<FileLink> {


  protected boolean searchFileName;

  protected boolean searchFileUri;

  protected boolean searchFileDescription;

  protected boolean inHtmlEmbeddableFilesOnly = false;


  public FilesSearch(String searchTerm, SearchCompletedListener<Collection<FileLink>> completedListener) {
    this(searchTerm, true, true, true, completedListener);
  }

  public FilesSearch(String searchTerm, boolean searchFileName, boolean searchFileUri, boolean searchFileDescription, SearchCompletedListener<Collection<FileLink>> completedListener) {
    super(searchTerm, completedListener);
    this.searchFileName = searchFileName;
    this.searchFileUri = searchFileUri;
    this.searchFileDescription = searchFileDescription;
  }


  public boolean searchFileName() {
    return searchFileName;
  }

  public boolean searchFileUri() {
    return searchFileUri;
  }

  public boolean searchFileDescription() {
    return searchFileDescription;
  }

  public boolean inHtmlEmbeddableFilesOnly() {
    return inHtmlEmbeddableFilesOnly;
  }

  public void setInHtmlEmbeddableFilesOnly(boolean inHtmlEmbeddableFilesOnly) {
    this.inHtmlEmbeddableFilesOnly = inHtmlEmbeddableFilesOnly;
  }

}

File: libs/service_layer/search/SearchEngine/src/main/kotlin/net/dankito/service/search/specific/FilesSearch.kt
package net.dankito.service.search.specific


import net.dankito.deepthought.model.FileLink
import net.dankito.service.search.Search
import net.dankito.service.search.SearchWithCollectionResult


class FilesSearch(searchTerm: String = Search.EmptySearchTerm,
                  val fileType: FileType = FileType.LocalOrRemoteFiles,
                  val searchUri: Boolean = true, val searchName: Boolean = true,
                  val searchMimeType: Boolean = true, val searchFileType: Boolean = true,
                  val searchDescription: Boolean = true, val searchSourceUri: Boolean = true,
                  completedListener: (List<FileLink>) -> Unit) : SearchWithCollectionResult<FileLink>(searchTerm, completedListener) {


    enum class FileType {
        LocalFilesOnly,
        RemoteFilesOnly,
        LocalOrRemoteFiles
    }

}
